climada/hazard/relative_cropyield.py

```diff
@@ -247,18 +252,15 @@ def set_rel_yield_to_int(self, hist_mean):
         idx = np.where(hist_mean != 0)[0]
 
         # initialize new hazard_matrix
-        hazard_matrix = np.empty(self.intensity.shape)
-        hazard_matrix[:, :] = np.nan
+        hazard_matrix = np.zeros(self.intensity.shape, dtype=np.float32)
 
         # compute relative yield for each event:
         for event in range(len(self.event_id)):
-            hazard_matrix[event, idx] = self.intensity[event, idx] / hist_mean[idx]
+            hazard_matrix[event, idx] = (self.intensity[event, idx] / hist_mean[idx])-1
 
         self.intensity = sparse.csr_matrix(hazard_matrix)
         self.intensity_def = 'Relative Yield'
         self.units = ''
-        self.intensity.max = np.nanmax(self.intensity.toarray())
-        self.intensity.min = np.nanmin(self.intensity.toarray())
 
         return self
 
```

Comment:
The double loop in `def set_percentile_to_int(self, reference_intensity=None):` can be simplified.